### PR TITLE
Improve cbr provider

### PR DESCRIPTION
**Pull request by remedge**
- add loader classes;
- add tests.

### CODE REVIEW

File: src/RedCode/Currency/Rate/XML/XMLParser.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+class XMLParser
+{
+    /**
+     * @param $string
+     * @return \SimpleXMLElement
+     */
+    public function parse($string)
+    {
+        return new \SimpleXMLElement($string);
```
Comment (maZahaca) [inline src/RedCode/Currency/Rate/XML/XMLParser.php]: What is the purpose to move it to external class?
You can reach the same effect just having SOAPLoader and mocking of it.

File: tests/CbrCurrencyRateProviderTest.php [hunk 2]
```diff
@@ -7,139 +7,175 @@
...
+        $this->assertEquals(3, count($rates));
+        foreach ($rates as $rate) {
             $this->assertInstanceOf('\\RedCode\\Currency\\Rate\\ICurrencyRate', $rate);
         }
+    }
+
+    /**
+     * @return SOAPLoader
+     */
+    private function _getSOAPLoaderMock()
```
Comment (maZahaca) [inline tests/CbrCurrencyRateProviderTest.php]: do not use "_" for private/protected methods

File: src/RedCode/Currency/Rate/XML/XMLParser.php (hunk 1 above)
Comment (remedge) [inline src/RedCode/Currency/Rate/XML/XMLParser.php]: You are quite right, it is unnecessary class.

File: tests/CbrCurrencyRateProviderTest.php (hunk 2 above)
Comment (remedge) [inline tests/CbrCurrencyRateProviderTest.php]: Fixed.

**Comment by maZahaca** [comment]
Great!

**Pull request merged**